openfl/component/aggregation_functions/weighted_average.py

```diff
@@ -2,21 +2,23 @@
 # SPDX-License-Identifier: Apache-2.0
 
 """Federated averaging module."""
+from typing import Any
+from typing import List
 
 import numpy as np
 
 from .core import AggregationFunction
 
 
-def weighted_average(tensors, weights):
+def weighted_average(tensors: np.ndarray, weights: np.ndarray) -> np.ndarray:
```

Comment:
A guess we shouldn't. As I understand we always keep tensor data as `np.ndarray`, isn't it?

Reply:
I see `tensors, weights = np.array(tensors), np.array(weights)` construction in the geometric_median.py. So it confused me. I think we should add this line, not only for proper typings, but according to [this](https://numpy.org/doc/stable/reference/generated/numpy.average.html) -  There will be an implicit conversion.
```
Array containing data to be averaged. If a is not an array, a conversion is attempted.
```

Reply:
This conversion is redundant here, since `np.average` function receives all array-like objects, including `list`, and converts them to NumPy arrays if needed.

Reply:
@itrushkin , we just have to convert it since we specify type such as `np.ndarray`. Yes, we can define `array_like` type, but it seems to me that this is not necessary, because inside the function it expects exactly `ndarray` (and will convert if it is not).

Reply:
No need to define `array_like` type. We can use [numpy.typing.ArrayLike](
https://numpy.org/devdocs/reference/typing.html#numpy.typing.ArrayLike).
Wider type makes function usable outside this module. We might pass any   sequence object to it and it would work.